src/google-maps/map-heatmap-layer/map-heatmap-layer.ts

```diff
@@ -0,0 +1,175 @@
+} from '@angular/core';
+
+import {GoogleMap} from '../google-map/google-map';
+
+/** Possible data that can be shown on a heatmap layer. */
+export type HeatmapData =
+  google.maps.MVCArray<
+    google.maps.LatLng | google.maps.visualization.WeightedLocation | google.maps.LatLngLiteral> |
+  google.maps.LatLng[] |
+  google.maps.LatLngLiteral[] |
+  google.maps.visualization.WeightedLocation[];
+
+
+/**
+ * Angular directive that renders a Google Maps heatmap via the Google Maps JavaScript API.
+ *
+ * See: https://developers.google.com/maps/documentation/javascript/reference/visualization
+ */
+@Directive({
+  selector: 'map-heatmap-layer',
+  exportAs: 'mapHeatmapLayer',
+})
+export class MapHeatmapLayer implements OnInit, OnChanges, OnDestroy {
+  /**
+   * Data shown on the heatmap.
+   * See: https://developers.google.com/maps/documentation/javascript/reference/visualization
+   */
+  @Input()
+  set data(data: HeatmapData) {
+    this._data = data;
+  }
+  private _data: HeatmapData;
+
+  /**
+   * Options used to configure the heatmap. See:
+   * developers.google.com/maps/documentation/javascript/reference/visualization#HeatmapLayerOptions
+   */
+  @Input()
+  set options(options: Partial<google.maps.visualization.HeatmapLayerOptions>) {
+    this._options = options;
+  }
+  private _options: Partial<google.maps.visualization.HeatmapLayerOptions>;
+
+  /**
+   * The underlying google.maps.visualization.HeatmapLayer object.
+   *
+   * See: https://developers.google.com/maps/documentation/javascript/reference/visualization
+   */
+  heatmap?: google.maps.visualization.HeatmapLayer;
+
+  constructor(
+    private readonly _googleMap: GoogleMap,
+    private _ngZone: NgZone) {}
+
+  ngOnInit() {
+    if (this._googleMap._isBrowser) {
+      if (!window.google?.maps?.visualization && (typeof ngDevMode === 'undefined' || ngDevMode)) {
+        throw Error(
+            'Namespace `google.maps.visualization` not found, cannot construct heatmap. ' +
+            'Please install the Google Maps JavaScript API with the "visualization" library: ' +
+            'https://developers.google.com/maps/documentation/javascript/visualization');
+      }
+
+      // Create the object outside the zone so its events don't trigger change detection.
+      // We'll bring it back in inside the `MapEventManager` only for the events that the
+      // user has subscribed to.
+      this._ngZone.runOutsideAngular(() => {
+        this.heatmap = new google.maps.visualization.HeatmapLayer(this._combineOptions());
+      });
+      this._assertInitialized();
+      this.heatmap.setMap(this._googleMap.googleMap!);
+    }
+  }
+
+  ngOnChanges(changes: SimpleChanges) {
+    const {_data, heatmap} = this;
+
+    if (heatmap) {
+      if (changes['options']) {
+        heatmap.setOptions(this._combineOptions());
+      }
+
+      if (changes['data'] && _data !== undefined) {
+        // TODO(crisbeto): the API allows for `MVCArray<LatLng|WeightedLocation>` or
```

Comment:
Is it worth sending a PR to fix the types?

Reply:
Yeah, I was planning on sending out out, but I haven't gotten around to it yet.

Reply:
Here we go https://github.com/DefinitelyTyped/DefinitelyTyped/pull/50423.